include/engine/geospatial_query.hpp

```diff
@@ -348,6 +348,56 @@ template <typename RTreeT, typename DataFacadeT> class GeospatialQuery
                 coordinates->at(data.u), coordinates->at(data.v), input_coordinate,
                 point_on_segment, ratio);
 
+        const auto translateAngleToLocation = [](const double angle)
```

Comment:
That proposal does not result in the same information that I want to have in the search.
                      A
              
***************        would result in `Ahead`, but should be `Left`, same with `Behind`.

If the ratio thing is ignored,
                                                  A
***************
would result in `Left` instead of `Ahead`.

My proposal how to handle this and other conflicts down the line remains to push the coordinate through the phantom node and do the computations in assemble_steps.

This would require a fix in the hint, apparently, since it is computed 1:1 from the phantom node. In my view, it should not include this information, if we have to pass it via the phantom node.

If we want to fix unnecessary segments during post-processing (snapped to the wrong phantom node at an intersection), we require the coordinate for correct calculations.